types/xmlhttprequest/index.d.ts

```diff
@@ -0,0 +1,8 @@
+// Type definitions for xmlhttprequest 1.8
+// Project: https://github.com/driverdan/node-XMLHttpRequest
+// Definitions by: Harry Cruse <https://github.com/CruseCtrl>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type XHR = XMLHttpRequest;
```

Comment:
Should you pick only some of the methods? The docs says it's based on some older XmlHttpRequest specs (v1)

Reply:
That's possible, but I didn't want to have to take the time to work out which functions are included and which ones aren't. I thought that it's better to have something in Definitely Typed than not have anything, and we can always improve them in the future. The [npm package](https://www.npmjs.com/package/xmlhttprequest) was last published 8 years ago and has 1.3M weekly downloads, and no one has bothered to create the type definitions for it so far, so hopefully this will pave the way for other people improving it if they find any issues.

Is that acceptable, or can we only merge something that we know to be perfectly accurate?

Reply:
Generally, partly accurate types are better, but I'm not sure for something so widely used. How have people been using it in Typescript thus far? Have they just been ignoring any errors when importing, or turning off `strict`?

If the new type is a superset of the old type -- that is, if the methods that existed in the v1 spec haven't changed their types -- then I think it's OK to include some methods that aren't actually available. Is that the case?

Reply:
I'm not entirely sure to be honest. I've inherited a project with `strict` turned off and which uses `xmlhttprequest` only in a single file. In the process of turning on `strict` I ended up creating a declaration file for `xmlhttprequest`, so I thought I might as well commit it here so that other people can use it.

I haven't actually written any of the code which uses this package, but here's roughly what it's doing:

```ts
import { XhrFactory } from '@angular/common';
import { XMLHttpRequest } from 'xmlhttprequest';

export class BrowserXhr implements XhrFactory {
  public build(): XMLHttpRequest {
    return new XMLHttpRequest();
  }
}
```

And after looking up [XhrFactory](https://angular.io/api/common/XhrFactory), it looks like it does need to be a full `XMLHttpRequest`, but I don't know whether that's V1 or V2. Presumably the code has been working fine up until this point though, so that leads me to believe that the types must be reasonably good

Reply:
That's good enough for me. @peterblazejewicz what do you think?

Reply:
Yes, let's have it

Reply:
The code in example, is just Angular factory to write a thing client doing Http request.

Reply:
Great, thanks for that!